pit30m/data/log_reader.py

```diff
@@ -75,17 +75,53 @@ def __init__(
             wgs84_pose_fname: Name of the WGS84 pose file (ie, global coords). This is usually "wgs84.npz.lz4"
             map: Map object. If not provided, will try to load it from the log root
             index_version: Version of the index to use. Currently only 0 is supported.
+            partitions: Set of partitions to load. These are used to load subset of the date (e.g., training queries).
+                defaults to None, which means that no sensor measurements are filtered.
         """
         self._log_root_uri = log_root_uri.rstrip("/")
         self._pose_fname = pose_fname
         self._wgs84_pose_fname = wgs84_pose_fname
         self._map = Map() if map is None else map
         # TODO(julieta) Semantic version this
         self._index_version = index_version
+        self.partitions = set() if partitions is None else partitions
 
     def __repr__(self) -> str:
         return f"Pit30M Log Reader: {self._log_root_uri}"
 
+    @cached_property
+    def partitions_index(self) -> np.ndarray:
+        """
+        Returns a boolean np array that accounts for the requested partitions, by setting sensor readings that should
+        be skipped to False. Currently computed from the Front Camera.
+        """
+        if not self.partitions:
+            n_sensor_measurements = len(self.get_cam_geo_index(CamName.MIDDLE_FRONT_WIDE))
+            return np.full(n_sensor_measurements, True)
+
+        partition_indices = self.partition_assigments
+        combined_indices = np.logical_and.reduce(partition_indices)
+        return combined_indices
+
+    @property
```

Comment:
cache?

Reply:
Yep! Added in [5b75a33](https://github.com/pit30m/pit30m/pull/35/commits/5b75a337ea74282ce5896d3a659e916b2c39b22a)